client-datastore-cassandra/src/main/java/com/telecominfraproject/wlan/client/datastore/cassandra/ClientDAO.java

```diff
@@ -378,6 +411,91 @@ public Client delete(int customerId, MacAddress clientMac) {
         LOG.debug("get({}, {}) returns {} record(s)", customerId, clientMacSet, results.size());
         return results;
     }
+    
+    public PaginationResponse<Client> searchByMacAddress(int customerId, String macSubstring,
+    		List<ColumnAndSort> sortBy, PaginationContext<Client> context) {
+        LOG.debug("calling searchByMacAddress({}, {})", customerId, macSubstring);
+
+        PaginationResponse<Client> ret = new PaginationResponse<>();
+        ret.setContext(context.clone());
+
+        if (ret.getContext().isLastPage()) {
+            // no more pages available according to the context
+            LOG.debug(
+                    "No more pages available when looking up Clients for customer {} and macSubstring {} with last returned page number {}",
+                    customerId, macSubstring, context.getLastReturnedPageNumber());
+            return ret;
+        }
+
+        LOG.debug("Looking up Clients for customer {} and macSubstring {} with last returned page number {}", 
+                customerId, macSubstring, context.getLastReturnedPageNumber());
+
+        // add sorting options for the query
+        // Cassandra allows very limited support for ordering results
+        // See https://cassandra.apache.org/doc/latest/cql/dml.html#select
+        // In here allowed orderings are the order induced by the clustering columns and the reverse of that one.
+        // also, order by with secondary indexes is not supported
+        // ***** We will ignore the order supplied by the caller for this datastore
+        
+        ArrayList<Object> bindVars = new ArrayList<>();
+        BoundStatement boundStmt;
+        
+        if (macSubstring != null) {
+		    bindVars.add(customerId);
+		    bindVars.add("%" + macSubstring.toLowerCase() + "%");
+	                
+	        boundStmt = preparedStmt_getClientMacLikeMacString.bind(bindVars.toArray());
+        } else {
+		    bindVars.add(customerId);
+	                
+	        boundStmt = preparedStmt_getClientMacByCustomerId.bind(bindVars.toArray());
```

Comment:
this line could go below the if() - it happens in both if and else

Reply:
The 2 lines are using different PreparedStatements, one without the substring condition and one with the substring. 

Did you mean to get the proper statement in the if() and then bind them outside the if()?

Reply:
You are right, those are two different statements, and they should stay as they are. 
Sorry, my bad.